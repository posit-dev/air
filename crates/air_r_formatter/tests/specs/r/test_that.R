test_that("description", {
	1 + 1
})

extended_test_that("description", {
	1 + 1
})

test_that("description that is super long and actually exceeds the line limit but we arent going to break!", {
	1 + 1
})

test_that(desc = "description that is super long and actually exceeds the line limit but we arent going to break!", code = {
	1 + 1
})

# Opening brace is moved back onto the first line because this is a test call
test_that("description that is super long and actually exceeds the line limit but we arent going to break!",
{
	1 + 1
})

# Both arguments are reflowed because this is a test call
test_that(
  "description that is super long and actually exceeds the line limit but we arent going to break!", {
	1 + 1
})

# The first argument isn't a string, so this isn't special cased and
# it gets expanded
# TODO: The expansion doesn't look right though
test_that(identity("description that is super long and actually exceeds the line limit but we arent going to break!"), {
	1 + 1
})
