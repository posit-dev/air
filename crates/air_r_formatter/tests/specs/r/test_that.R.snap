---
source: crates/air_formatter_test/src/snapshot_builder.rs
info: r/test_that.R
---
# Input

```R
test_that("description", {
	1 + 1
})

extended_test_that("description", {
	1 + 1
})

test_that("description that is super long and actually exceeds the line limit but we arent going to break!", {
	1 + 1
})

test_that(desc = "description that is super long and actually exceeds the line limit but we arent going to break!", code = {
	1 + 1
})

# Opening brace is moved back onto the first line because this is a test call
test_that("description that is super long and actually exceeds the line limit but we arent going to break!",
{
	1 + 1
})

# Both arguments are reflowed because this is a test call
test_that(
  "description that is super long and actually exceeds the line limit but we arent going to break!", {
	1 + 1
})

# The first argument isn't a string, so this isn't special cased and
# it gets expanded
# TODO: The expansion doesn't look right though
test_that(identity("description that is super long and actually exceeds the line limit but we arent going to break!"), {
	1 + 1
})

```


=============================

# Outputs

## Output 1

-----
Indent style: Space
Indent width: 2
Line ending: LF
Line width: 80
-----

```R
test_that("description", {
  1 + 1
})

extended_test_that("description", {
  1 + 1
})

test_that("description that is super long and actually exceeds the line limit but we arent going to break!", {
  1 + 1
})

test_that(desc = "description that is super long and actually exceeds the line limit but we arent going to break!", code = {
  1 + 1
})

# Opening brace is moved back onto the first line because this is a test call
test_that("description that is super long and actually exceeds the line limit but we arent going to break!", {
  1 + 1
})

# Both arguments are reflowed because this is a test call
test_that("description that is super long and actually exceeds the line limit but we arent going to break!", {
  1 + 1
})

# The first argument isn't a string, so this isn't special cased and
# it gets expanded
# TODO: The expansion doesn't look right though
test_that(
  identity(
    "description that is super long and actually exceeds the line limit but we arent going to break!"
  ),
  {
    1 + 1
  }
)
```

# Lines exceeding max width of 80 characters
```
    9: test_that("description that is super long and actually exceeds the line limit but we arent going to break!", {
   13: test_that(desc = "description that is super long and actually exceeds the line limit but we arent going to break!", code = {
   18: test_that("description that is super long and actually exceeds the line limit but we arent going to break!", {
   23: test_that("description that is super long and actually exceeds the line limit but we arent going to break!", {
   32:     "description that is super long and actually exceeds the line limit but we arent going to break!"
```
